app/scripts/HiGlassComponent.js

```diff
@@ -4075,53 +4075,82 @@ class HiGlassComponent extends React.Component {
+              );
+
+              this.setCenters[viewUid](centerX, centerY, k, false, animateTime);
+            },
+            this.pubSub,
+          );
+        } else {
+          console.warn(`Couldn't find the gene symbol: ${gene}`);
+        }
+      }
+    });
+  }
+
+  suggestGene(viewUid, keyword, callback) {
+    if (!(viewUid in this.setCenters)) {
+      throw Error(
+        `Invalid viewUid. Current uuids: ${Object.keys(this.setCenters).join(
+          ',',
+        )}`,
+      );
+    }
+
+    if (
+      !this.state.views[viewUid].genomePositionSearchBox ||
+      !this.state.views[viewUid].genomePositionSearchBox.autocompleteServer ||
+      !this.state.views[viewUid].genomePositionSearchBox.autocompleteId
+    ) {
+      console.warn(
+        'Please set autocompleteServer and autocompleteId to use the suggestGene API',
+      );
+      return;
+    }
+
     const autocompleteServer = this.state.views[viewUid].genomePositionSearchBox
       .autocompleteServer;
     const autocompleteId = this.state.views[viewUid].genomePositionSearchBox
       .autocompleteId;
-    const chromInfoPath = this.state.views[viewUid].chromInfoPath;
 
-    const url = `${autocompleteServer}/suggest/?d=${autocompleteId}&ac=${geneSymbol.toLowerCase()}`;
+    const url = `${autocompleteServer}/suggest/?d=${autocompleteId}&ac=${keyword.toLowerCase()}`;
 
     tileProxy
       .json(url, toVoid, this.pubSub)
-      .then((positions) => {
-        if (positions) {
-          // extract the position of exact match
-          const genePosition = positions.find(
-            (d) => d.geneName.toLowerCase() === geneSymbol.toLowerCase(),
-          );
-
-          if (genePosition) {
-            const { chr, txStart: start, txEnd: end } = genePosition;
-
-            // extract absolute positions
-            ChromosomeInfo(
-              chromInfoPath,
-              (loadedChromInfo) => {
-                // using the absolution positions, zoom to the position near a gene
-                const startAbs = loadedChromInfo.chrToAbs([chr, start]);
-                const endAbs = loadedChromInfo.chrToAbs([chr, end]);
-
-                const [centerX, centerY, k] = scalesCenterAndK(
-                  this.xScales[viewUid].copy().domain([startAbs, endAbs]),
-                  this.yScales[viewUid].copy().domain([startAbs, endAbs]),
-                );
-
-                this.setCenters[viewUid](
-                  centerX,
-                  centerY,
-                  k,
-                  false,
-                  animateTime,
-                );
-              },
-              this.pubSub,
-            );
-          } else {
-            console.warn(`Couldn't find the gene symbol: ${geneSymbol}`);
-          }
-        }
+      .then((suggestions) => {
+        callback(
+          suggestions.map((d) => {
+            // change to use simpler key names just to make it easier to use
```

Comment:
I don't think this is a good idea. It's introducing an inconsistency between the data returned from the server API and client API.

Not worth making it "easier to use", IMO. Also, I don't necessarily agree that it's easier to use. `txStart` and `geneName` convey more information than `start` and `gene`.

Reply:
Thanks for pointing this out. I guess I was trying to over-simplify the variable names, and I agree that it is not a good idea to introduce the inconsistency. I've renamed the variables to use the names as they were.